docs: add request reviews use case (#2419)
Mergifyio/docs: src/content/workflow/request-reviews.mdx

---
title: Request Reviews
description: Enhance your review workflow with customized, efficient, and dynamic review assignments tailored to your needs.
---

Mergify's [`request_reviews`](/workflow/actions/request_reviews) action
provides an enhanced mechanism to automate and personalize your review process,
taking it several notches above GitHub's conventional review assignment. By
harnessing the versatility of pull request rules and conditions, you can define
intricate workflows tailored to your team's needs. Let's explore how to utilize
the action for various scenarios.

`video: https://www.youtube.com/watch?v=eIrdsALyVJQ&t=2s`

Below is a list of examples of what you could achieve with pull request rules
and the `request_reviews` action.

## Basic Assignment

Automatically request specific users or teams to review a pull request when
it's labeled with `review`. The rule checks for a pull request labeled `review`
and then requests reviews from `user1`, `user2`, and `team1`.

```yaml
pull_request_rules:

>   - name: request review from user and team when pull request is labeled with `review`

Reviewer: ```suggestion
  - name: request review from users and team when pull request is labeled with `review`
```

    conditions:
      - label=review
    actions:
      request_reviews:
        users:
          - user1
          - user2
        teams:
          - team1
```

## Flexible Reviewers Assignment

> Dynamic reviewer assignments based on the content and context of pull requests.

Reviewer: ```suggestion
Dynamic reviewers assignment based on the content and context of pull requests.
```

### File Type Based Review

Assign a Python expert to review changes in Python files.

```yaml
pull_request_rules:
  - name: ask jd to review changes on python files
    conditions:
      - files~=\.py$
    actions:
      request_reviews:
        users:
          - jd
```

### Label Based Review

For security-focused pull requests, you might want specialized teams to review.

```yaml
pull_request_rules:

>   - name: ask the security team to review security labeled PR

Reviewer: ```suggestion
  - name: ask the security teams to review security labeled PR
```
